Sources/MongoSwift/Operations/Operation.swift

```diff
@@ -87,28 +88,31 @@ internal class OperationExecutor {
         if let session = session {
             // start the session if needed (which generates a new operation itself), and then execute the operation.
             return session.startIfNeeded()
-                .flatMap { self.execute(doOperation) }
+                .flatMap { self.execute(on: eventLoop, doOperation) }
                 .flatMap { resubmitIfNeeded($0) }
         }
 
         // no session was provided, so we can just jump to executing the operation.
-        return self.execute(doOperation).flatMap { resubmitIfNeeded($0) }
+        return self.execute(on: eventLoop, doOperation).flatMap { resubmitIfNeeded($0) }
     }
 
-    internal func execute<T>(_ body: @escaping () throws -> T) -> EventLoopFuture<T> {
-        self.threadPool.runIfActive(eventLoop: self.eventLoopGroup.next(), body)
+    internal func execute<T>(on eventLoop: EventLoop? = nil, _ body: @escaping () throws -> T) -> EventLoopFuture<T> {
+        self.threadPool.runIfActive(eventLoop: eventLoop ?? self.eventLoopGroup.next(), body)
     }
 
-    internal func makeFailedFuture<T>(_ error: Error) -> EventLoopFuture<T> {
-        self.eventLoopGroup.next().makeFailedFuture(error)
+    internal func makeFailedFuture<T>(_ error: Error, on eventLoop: EventLoop? = nil) -> EventLoopFuture<T> {
```

Comment:
The default value of nil for the `eventLoop` is only temporarily so Swift does not complain about missing arguments. These will be removed once the helpers are introduced in the upcoming PRs.